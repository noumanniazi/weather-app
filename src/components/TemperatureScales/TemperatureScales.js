// lib
import React from "react";
import PropTypes from "prop-types";
import Radio from "antd/lib/radio";

const TemperatureScales = props => {
  const { currentScale, setCurrentScale } = props;
  return (
    <div className="temperature-scale">
      <Radio.Group
        onChange={e => {
          setCurrentScale(e.target.value);
        }}
        value={currentScale}
      >
        <Radio value={"C"}>Celsius</Radio>
        <Radio value={"F"}>Fahrenheit</Radio>
      </Radio.Group>
    </div>
  );
};

TemperatureScales.prototype = {
  currentScale: PropTypes.string.isRequired,
  setCurrentScale: PropTypes.func.isRequired
};

export default TemperatureScales;
