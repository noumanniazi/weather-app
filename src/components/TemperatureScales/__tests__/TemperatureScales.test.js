// lib
import React from "react";
import renderer from "react-test-renderer";

// src
import TemperatureScales from "../TemperatureScales";

describe("Testing TemperatureScales Component", () => {
  it("Matches TemperatureScales snapshot", () => {
    const data = {
        "currentScale": "C",
        setCurrentScale: ()=>{}
      };
    const temperatureScales = renderer.create(<TemperatureScales {...data} />);
    let tree = temperatureScales.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
