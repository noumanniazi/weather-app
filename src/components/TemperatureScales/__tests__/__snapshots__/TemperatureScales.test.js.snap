// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TemperatureScales Component Matches TemperatureScales snapshot 1`] = `
<div
  className="temperature-scale"
>
  <div
    className="ant-radio-group ant-radio-group-outline"
  >
    <label
      className="ant-radio-wrapper ant-radio-wrapper-checked"
    >
      <span
        className="ant-radio ant-radio-checked"
        style={Object {}}
      >
        <input
          checked={true}
          className="ant-radio-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="C"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        Celsius
      </span>
    </label>
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="F"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        Fahrenheit
      </span>
    </label>
  </div>
</div>
`;
